### DISCUSSION TITLE
DOOM

### DISCUSSION THREAD

**Discussion by ghost** [janet-lang/janet#1450] [Ideas]
_(I would have put this on either live space but they're both borked for me.)_

I use DOOM Emacs, which is pretty good except it's a tad...clunky. I don't _notice_ the latency but I do _feel_ it, and I don't even try to understand what's going on under the hood; but to my knowledge there isn't a single other environment that can do all the things it does and integrate them so well without having to write half the logic myself.

A big negative is, of course, the lack of a Janet module. I'm sure I could finagle `janet-emacs-trial-kit` into the DOOM framework, but of course there'd be no Evil bindings and that's kind of a Shellstrop. So, I'm now _thinking_ of writing such a module myself. I can't guarantee it won't slip over the Horizon like all my other insane ideas, I _can_ guarantee that if it doesn't it won't materialise for a _long_ time, and really I'm aware I have no relevant qualifications and plenty of relevant *dis*qualifications. But hopefully by making this one known, I might actually get around to it.

**Comment by sogaiu** [comment]
@welcome-linja I think @llmII may be using Doom Emacs with some Janet support.  May be he might have some pointers and/or tips.

I think at one point his setup used [janet-ts-mode](https://github.com/sogaiu/janet-ts-mode) (which in turn requires Emacs >= 29.1 with the integrated tree-sitter support [1]).

---

[1] janet-emacs-trial-kit also uses janet-ts-mode FWIW.

**Reply by ghost** [reply]
Cool. I have some time to wait for him to see that since I just submitted my only dev-capable machine for keyboard repair. Cool.

**Reply by sogaiu** [reply]
May the gods of repair smile kindly on your hardware.
